src/main/core/Subscriber/SecurityEventSubscriber.php

```diff
@@ -48,18 +50,40 @@ public static function getSubscribedEvents(): array
             SecurityEvents::VIEW_AS => 'logEvent',
             SecurityEvents::VALIDATE_EMAIL => 'logEvent',
             SecurityEvents::AUTHENTICATION_FAILURE => 'logEvent',
+            SecurityEvents::SWITCH_USER => 'logEventSwitchUser',
         ];
     }
 
     public function logEvent(Event $event, string $eventName): void
     {
         $logEntry = new SecurityLog();
         $logEntry->setDetails($event->getMessage($this->translator));
-        $logEntry->setEvent($eventName);
+        $logEntry->setEvent($this->translator->trans($eventName, [], 'security'));
```

Comment:
You should store the real event and do the translation in the ui.

Reply:
An example of translated col in ListData : https://github.com/claroline/Claroline/blob/13.x/src/main/core/Resources/modules/modals/template-types/components/modal.jsx#L34